mantidproject/mantid
# Add NotFinite to MaskDetectorsIf and make it check every bin

**Description of work.**
Three changes:
1. Added a Notfinite option to the operator property
1. changed the alg to test every bin rather than just the first bin
1. Added a "\n" to the warning if not output workspace is set, as otherwise it was not appearing.
1. Left a large comment in place where the alg is not applying masking correctly.  Not changing at the moment as this has been added to the TSC agenda to decide the correct approach.

**To test:**
``` python
from mantid.simpleapi import *
ws = Load("MAR11060")
for i in range(100,150):
    ws.dataY(i)[2] = 'NaN'

ws_fix = ReplaceSpecialValues(ws,NaNValue = -111, Infinityvalue = -111)
MaskDetectorsIf(ws_fix, Value = -111) #this should give a visible warning message and do nothing
ws_fix2 = MaskDetectorsIf(ws_fix, Value = -111)
ws_fix3 = MaskDetectorsIf(ws, Operator="NotFinite") 
```
1. Look at the workspaces, fix 2 and 3 should be equivalent. Spectra 100-150 should be masked.
1. Before the changes, Fix 2 add 3 would not have any masked spectra.
re #27642. 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
